PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/resources/META-INF/ruleConfigs.json

```diff
@@ -10,5 +10,11 @@
   "ServiceBusReceiverAsyncClientCheck": {
     "client_name": "ServiceBusReceiverAsyncClient",
     "antipattern_message": "Use of ServiceBusReceiverAsyncClient detected. Use ServiceBusProcessorClient instead."
+  },
+
+  "HardcodedAPIKeysAndTokensCheck": {
+    "clients_to_check": ["AzureKeyCredential", "AccessToken"],
```

Comment:
Consider adding `KeyCredential`, `AzureNamedKeyCredential`, `AzureSasCredential`, `AzureNamedKey`, `ClientSecretCredential`, `UsernamePasswordCredential`, and `BasicAuthenticationCredential`. All of these are similar in that they might encourage hard coding a secret in some form.

Reply:
done:) these 2 use builders for instantiation so I'm checking for the builders: ClientSecretCredentialBuilder, "UsernamePasswordCredentialBuilder